kuma/api/v1/tests/test_views.py

```diff
@@ -330,3 +330,42 @@ def test_search_basic(self):
         assert data['documents']
         assert data['count'] == 5
         assert data['locale'] == 'fr'
+
+
+@pytest.mark.parametrize(
+    'http_method', ['put', 'post', 'delete', 'options', 'head'])
+def test_get_user_disallowed_methods(client, wiki_user, http_method):
+    """
+    HTTP methods other than GET are not allowed on the api.v1.get_user endpoint.
+    """
+    url = reverse('api.v1.get_user', args=(wiki_user.username,))
+    response = getattr(client, http_method)(url)
+    assert response.status_code == 405
+    assert_no_cache_header(response)
+
+
+@pytest.mark.parametrize('case', ('upper', 'lower'))
```

Comment:
I'm not sure this is worth it. There is no way we can make it NOT case-insensitive (double-negative) so by doing this you're basically unit testing the MySQL collation. This testing isn't testing your code, but testing implementation details in the system.

Reply:
My thought here is that this is testing the "contract" of the endpoint, and case insensitivity is part of that contract, no matter how it's implemented. I would say it's important that the tests detect a change in that "contract", so for example, if we inadvertently modified the MySQL collation or we moved to PostgreSQL and didn't account for the collations.